server/src/main/java/org/opensearch/search/wlm/QueryGroupLevelResourceUsageView.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.search.wlm;
+
+import org.opensearch.common.annotation.ExperimentalApi;
+import org.opensearch.search.ResourceType;
+import org.opensearch.tasks.Task;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * Represents the point in time view of resource usage of a QueryGroup and
+ * has a 1:1 relation with a QueryGroup.
+ * This class holds the QueryGroup ID, the resource usage data, and the list of active tasks.
+ */
+@ExperimentalApi
+public class QueryGroupLevelResourceUsageView {
+
+    private final String queryGroupId;
```

Comment:
Do we need queryGroupId here? Considering this class intention is to just store resourceUsage and tasks list for simplification?
I see you are using it in equals/hashcode but why do we need that? When are we going to compare two resourceView objects?